### ISSUE TITLE
[BUG] openapi-generator-online require environment variable to define scheme/host/port in generated download links

### ISSUE THREAD

**Issue by jimschubert** [OpenAPITools/openapi-generator#3288]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

The `GENERATOR_HOST` environment variable in openapi-generator-online isn't necessary and can't accurately be set in some cases.

##### openapi-generator version

all?

##### Command line used for generation

n/a

##### Steps to reproduce

* Start a docker image and bind all exposed ports to random ports on the host:
```
docker run -it -P --rm openapitools/openapi-generator-online:v4.0.2
```
* Observe: `GENERATOR_HOST` would need to be passed at `docker run`, which isn't possible when docker randomizes port assignments. 
See 
<img width="1433" alt="image" src="https://user-images.githubusercontent.com/109659/60761330-08446180-a014-11e9-81d2-9ec75479ae26.png">

##### Related issues/PRs

#3287

##### Suggest a fix

Remove `GENERATOR_HOST` and pull scheme/host/port from request details.

`GENERATOR_HOST` itself adds some other complexity, for example if a single nginx configuration is used to proxy multiple addresses to the same instance of the service (e.g. `api.example.com`, `example.com`, and `latest.example.com` could point to the same service via single nginx configuration, but would all require separate processes due to `GENERATOR_HOST`.

**Comment by jimschubert** [comment]
I found that this condition is really only a problem when running from within a docker container. Setting the default value to an empty string appears to fix the bug. Running the service from command line or other means without the `GENERATOR_HOST` variable also works as expected. I opened #3289 to fix and add some documentation around the usage of the variable.

**Comment by jimschubert** [comment]
Resolved by #3289 

```
➜  openapi git:(master) docker run -it -P  -d --rm openapitools/openapi-generator-online:latest
7405816154d9a62f6b68ffcce4004a6d8ee34084e0aae0e55806c8913705e835
➜  openapi git:(master) ➜  openapi git:(master) curl -s \
    -X POST "http://localhost:32776/api/gen/servers/go-server" \
    -H "Accept: */*" \
    -H "Content-Type: application/json" \
    -d "{ \"openAPIUrl\": \"https://raw.githubusercontent.com/OpenAPITools/openapi-generator/master/modules/openapi-generator/src/test/resources/2_0/petstore.yaml\"}" \
   | jq .
{
  "code": "176970e4-ad32-49d5-8bff-7f7be2c89be4",
  "link": "http://localhost:32776/api/gen/download/176970e4-ad32-49d5-8bff-7f7be2c89be4"
}
```

Note the `http://localhost:32776` in the curl POST and the response's generated link location.
